packages/apps/dev-wallet/src/pages/create-profile/create-profile.tsx

```diff
@@ -1,34 +1,58 @@
 import { useHDWallet } from '@/modules/key-source/hd-wallet/hd-wallet.hook';
 import { useNetwork } from '@/modules/network/network.hook';
 import { IKeySource } from '@/modules/wallet/wallet.repository';
+import {
+  authCard,
+  backBtnClass,
+  buttonClass,
+  iconStyle,
+  inputClass,
+} from '@/pages/create-profile/create-profile.css.ts';
 import { kadenaGenMnemonic } from '@kadena/hd-wallet';
-import { Box, Button, Heading, Text, TextField } from '@kadena/react-ui';
-import { useState } from 'react';
+import { MonoArrowBackIosNew } from '@kadena/react-icons';
+import { Box, Button, Heading, Stack, Text, TextField } from '@kadena/react-ui';
+import { useRef, useState } from 'react';
 import { useForm } from 'react-hook-form';
-import { Navigate } from 'react-router-dom';
+import { Link, Navigate } from 'react-router-dom';
 import { useWallet } from '../../modules/wallet/wallet.hook';
 
 export function CreateProfile() {
   const { register, handleSubmit } = useForm<{
     password: string;
     profileName: string;
+    confirmation: string;
   }>();
   const { createProfile, isUnlocked, createKey, createKAccount } = useWallet();
   const { activeNetwork } = useNetwork();
   const [createdKeySource, setCreatedKeySource] = useState<IKeySource>();
   const { createHDWallet } = useHDWallet();
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState<boolean>(true);
+  const passwordRef = useRef<HTMLInputElement>(null);
+  const passwordConfirmationRef = useRef<HTMLInputElement>(null);
+  const validate = (value: string) => {
```

Comment:
you don't need these refs and the validate function.
you can use `react-hook-form` features
https://react-hook-form.com/docs/useform#mode